libraries/chain/webassembly/cf_transaction.cpp

```diff
@@ -6,24 +6,35 @@ namespace eosio { namespace chain { namespace webassembly {
       if( data.size() == 0 ) return transaction_size();
 
       const packed_transaction& packed_trx = context.trx_context.packed_trx;
-      const bytes& trx =
-            packed_trx.get_compression() == packed_transaction::compression_type::none ?
-               packed_trx.get_packed_transaction() :
-               fc::raw::pack( static_cast<const transaction&>( packed_trx.get_transaction() ) );
-
+      bytes trx = fc::raw::pack( static_cast<const transaction&>( packed_trx.get_transaction() ) );
```

Comment:
Just curious, what problem does this fix? Thanks.

Reply:
The packed transaction packed bytes stored in `packed_transaction::packed_trx` does not have to be the exact format that is returned by `fc::raw::pack` as long as it can be unpacked into a `transaction`. We store the original array of bytes in the `packed_transaction` however if a contract uses read_transaction it expects the format provided by `fc::raw::pack` as it might hash it to get the transaction id. So the optimization in the replaced code can't be used as the format might be different.

Reply:
Thanks Kevin for the detailed information!